### DISCUSSION TITLE
JSON line information output from .dat coverage

### DISCUSSION THREAD

**Discussion by dariuskia** [linux-test-project/lcov#300] [Q&A]
I've been using [bazel coverage](https://bazel.build/configure/coverage) to generate a combined coverage report `.dat` file, and then `genhtml`  to produce the viewable report. However, I'd like to have a parsed output of the lines that were hit for each test target, e.g. in JSON like [gcov's functionality](https://gcovr.com/en/stable/output/json.html). My intent is to produce a dataset of mappings between test targets and a structured (machine-readable) view of the lines that were hit. Is this possible? 

According to [this](https://stackoverflow.com/questions/27246076/does-lcov-generate-intermediate-gcov-files), `lcov` automatically erases the intermediate `.gcov` files, so `gcov -j` isn't an option. The most information I've been able to get is from `lcov -l`; however, this doesn't provide the exact list of lines that were hit.

**Comment by henry2cox** [comment]
I don't think I understand what you are looking for - though it sounds as if you want to use lcov as a driver - to run gcov for you.
If that is the case:  you `lcov --preserve --capture ...` might do something similar.
You probably also want to provide the temp directory - so you don't get a random one in /tmp.

More generally:  if your actual final application worked perfectly:  what did it do?
Not the data collection part - the actual application.  
There may be another way or an easier way to accomplish your goal.
